pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/QueryRunner.java

```diff
@@ -415,4 +399,31 @@ private PlanNode substituteNode(PlanNode node, Map<PlanNode, ? extends PlanNode>
       return node;
     }
   }
+
+  // Time series related utility methods below
```

Comment:
Moved the two existing methods here to make it easy to read.

Added the new `extractTimeoutMs` method too.